### PR TITLE
a11y improvements for site navigation

### PR DESCRIPTION
**Pull request by chancestrickland**
This PR makes a few minor but important adjustments for better accessibility.
- Fixed some invalid HTML in the list (missing `li` items)
- Added a proper button to the nav toggle with a visually hidden label
- Added a skip nav link for better keyboard/screen reader navigation

It also makes some small tweaks to the header so that the menu is toggled correctly when switching the screen between small and medium widths. Lastly, I broke out `PrimaryNav` from `Nav` so that `Nav` exports more generally useful nav menu components we can use elsewhere on the site.

Changes are not visual for the most part, so I'm not sure that a video or screens are helpful. The main exception being that when you tab from the top of the document, you will land on a "Skip to content" link that is otherwise hidden.

### CODE REVIEW

File: src/components/Nav.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import React, { forwardRef } from 'react';
+import { Flex, Link, Box, List, ListItem } from '@chakra-ui/core';
+import { Link as GatsbyLink } from 'gatsby';
+
+const Nav = forwardRef(
```
Comment (domitriusclark) [inline src/components/Nav.js:5]: TIL more about `forwardRef` 👍

File: src/components/Nav.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import React, { forwardRef } from 'react';
+import { Flex, Link, Box, List, ListItem } from '@chakra-ui/core';
```
Comment (domitriusclark) [inline src/components/Nav.js:2]: `Box` is unused here -- think this was just missed from initial

Comment (chancestrickland) [reply src/components/Nav.js]: @domitriusclark Removed

**Comment by magnificode** [comment]
Excellent work @chancestrickland - thank you!

**Pull request merged**